### ISSUE TITLE
Array out of bounds in base_reducer interpolate_wf_peak function

### ISSUE THREAD

**Issue by RichardWhite109** [cta-chec/CHECLabPy#28]
- Need to protect bounds when indexing in waveforms
- Peak finding will fail often / always when dealing with files containing no laser events
- If the tevent from the average across and event has a peak < 2 mV or outside 20 - 80 ns then could just set tevent to the middle of the window.. maybe not the best idea, but could make a 'better' reducer to be used specifically for noise analysis later

```
Traceback (most recent call last):
  File "/home/cta/Software/CHECLabPy/scripts/extract_dl1.py", line 131, in <module>
    main()
  File "/home/cta/Software/CHECLabPy/scripts/extract_dl1.py", line 94, in main
    params = reducer.process(waveforms_bs)
  File "/home/cta/Software/CHECLabPy/CHECLabPy/core/base_reducer.py", line 47, in process
    params.update(self._get_timing(waveforms))
  File "/home/cta/Software/CHECLabPy/CHECLabPy/core/base_reducer.py", line 137, in _get_timing
    t_pulse, amp_pulse = self.interpolate_wf_peak(waveforms)
  File "/home/cta/Software/CHECLabPy/CHECLabPy/core/base_reducer.py", line 123, in interpolate_wf_peak
    y2 = waveforms[self.pixel_arange, t1 + 1]
IndexError: index 128 is out of bounds for axis 1 with size 128
```

**Comment by watsonjj** [comment]
Yes, I will restrict t_event to be within 10 to (n_samples - 10) to avoid this problem, does that seem reasonable?

**Comment by RichardWhite109** [comment]
Yes - that sounds good. At a higher level we can add cuts - for example when making plots of FWHM there's no point in trying for pulses found with low amplitudes or with arrival times very far from the mean

**Comment by watsonjj** [comment]
Fixed in #31
